src/Schema/SQLServerSchemaManager.php

```diff
@@ -90,6 +90,7 @@ public function listSchemaNames(): array
 SELECT name
 FROM   sys.schemas
 WHERE  name NOT IN('guest', 'INFORMATION_SCHEMA', 'sys')
+AND    name NOT IN(SELECT name FROM sys.database_principals WHERE type='R')
```

Comment:
Does it have to be a sub-query? Additionally, the following query ([source](https://blog.quest.com/using-database-schemas-in-sql-server/)) looks more generic:
```sql
select s.name as schema_name, 
    s.schema_id,
    u.name as schema_owner
from sys.schemas s
    inner join sys.sysusers u
        on u.uid = s.principal_id
where u.issqluser = 1
    and u.name not in ('sys', 'guest', 'INFORMATION_SCHEMA')
```
Would it make sense to use it instead?

Reply:
I checked your proposal SQL and it will exclude different "database principal"s schemas from schema list ie. domain group/user owned schemas (issqluser=0)

I found document: [SQLCentral](https://www.sqlservercentral.com/scripts/list-schemas-in-a-database) and checked on databases I have access (v.2012-v.2022) and I think it is correct way to list it.
Finally condition: schema_id<16000 should be added to "where" section so it will looks like:
```sql
SELECT name
  FROM sys.schemas
 WHERE name NOT IN('guest', 'INFORMATION_SCHEMA', 'sys')
   AND schema_id<16000
```